src/main/scala/com/github/tototoshi/csv/CSVWriter.scala

```diff
@@ -69,7 +69,16 @@ class CSVWriter(protected val writer: Writer)(implicit val format: CSVFormat) ex
       quote _ compose escape compose show
     }
 
-    printWriter.print(fields.map(renderField).mkString(format.delimiter.toString))
+    val iterator = fields.iterator
+    var hasNext = iterator.hasNext
+    while (hasNext) {
+      printWriter.print(renderField(iterator.next()))
```

Comment:
It just came to me that we shouldn't render field to create a new string at all. We can write the rendered field directly to output. Should be faster and allocate less memory :)